### PR TITLE
Fix comma payment cart page

### LINKED ISSUE #24103: Adding payment with comma does not work anymore PS 1.7.7.x
### Describe the bug

From PS 1.7.7.x and up it is not more possible to add a payment with a "," (comma) in the amount, for example: 20,90 €
This is not very usefull for german Prestashop users. 

### Expected behavior

It's possible to add a payment with a "," (comma) in the amount

Adds a type class that governs the numeric type fields, checks that the expected format.

**specs**

The numeric type fields can accept the different number formats and convert it to the format managed by the PS system directly in the field. All non-numeric characters are ignored, except for the last one which is converted to a decimal point symbol.
If a non-numeric character is used twice, and it's the only one that is used, consider this character is a thousand separator.

**When the user unfocuses (blurs) the field, PS will convert the field following the rules describes above.**
 
1.000,2 => 1000.2
1N000,2 => 1000.2
100,2 -> 100.2
1,000,2 -> 10002
100,002 -> 100.002

List of the inputs in the BO:

_Order View_

* Amont in the tab payment
* Add a product (tax inclus and exclu) 
* Edit a product (tax inclus and exclu) 
* Add a discount (percentage and amount)

_Add Order_

* Add a discount
* Add a product

_Product page_

* Basic Tab: price TTC and HT

  _Combinaisons_

* Impact on price on the list 

  _Bulkaction_

* Cost price
* Impact on price (tax incl)
* Impact on price (tax excl)

_Shipping_

* Shipping fee
* Width
* Height
* Depth
* Weight

_Pricing_

* Price (tax excl)
* Price (tax incl)
* Price per unit (tax excl) 
* Cost Price (tax excl)

_Specific Price_

* Product price (tax excl.)
* Apply a discount of

_Discounts_

* Amount (tax incl and excl)
* Percentage

_Carriers_

* Shipping cost Ranges
* Width
* Height
* Depth
* Weight

_Carrier preference_

* Handling charges
* Free shipping starts at

_Taxes_

* Tax rate create & edit

_Order settings_

* Minimum purchase total required in order to validate the order 
* Gift-wrapping price 

#### Steps to Reproduce

1. Go to BO
2. Click on Orders > edit Order with "Awaiting payment" as status
3. Scroll down to **Payment** block and add a payment with comma (for example: 20,90 €)

**Screenshots**

image

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: N/A

### PR DESCRIPTION
**Pull request by prestaforum**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Change comma to dot for payment order
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #24103
| How to test?      | See #24103
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24179)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 1]
```diff
@@ -1161,7 +1161,7 @@ public function addPaymentAction(int $orderId, Request $request): RedirectRespon
                             $orderId,
                             $data['date'],
                             $data['payment_method'],
-                            $data['amount'],
+                            str_replace(',', '.', $data['amount']),
```
Comment (sowbiba) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:1164]: The comma can have different meanings depending in the context
https://docs.oracle.com/cd/E19455-01/806-0169/overview-9/index.html
replacing it like this is kinda ... rough

Comment (sowbiba) [reply src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:1164]: Maybe a DataTransformer in the form input which takes the context currency
Must be tried to know if it's possible

**Review by sowbiba — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php (hunk 1 above)
Comment (sowbiba) [reply src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:1164]: A discussion is in progress about this subject. I will let @PrestaShop/product-team give you the fix expected/planned

**Comment by marionf** [comment]
@MatShir I think this PR should rather be on 1.7.8.x branch wdyt ?

**Comment by Progi1984** [comment]
@prestaforum Could you check the PR from @PierreRambaud : #24339 ?

**Comment by PierreRambaud** [comment]
We will not merge a fix like that, we want to keep the backend safe and easy to maintain.
I'm closing it as we decided to do it only in FO.

**Pull request closed without merging**